apps_product_creator/models/product_product.py

```diff
@@ -88,15 +88,60 @@ def _manage_product_module(self, values):
         # Check if mandatory keys (for product module) are given
         if all([k in values_keys for k in check_keys]):
             product_obj = self.env['product.template']
+            module_version = self.env['odoo.module.version']
+            product_variant = self.env['product.product']
             product = product_obj.browse(values.get(product_tmpl_key, []))
             # If the product template is used to save an Odoo module
             if self._check_related_to_module(product):
                 attribute = self._get_version_attribute(values)
-                version = self._get_version_with_attribute(
+                versions = self._get_version_with_attribute(
                     product.odoo_module_id.module_version_ids, attribute)
+                for version in versions:
+                    module_id = version.module_id
+                    mod_ver_ids = module_id.dependence_module_version_ids.ids
+                    if version.module_id.dependence_module_version_ids:
```

Comment:
For reducing one indentation level and the cyclomatic complexity, put `if not version...: continue` instead